src/users/interfaces/challenge-response.interface.ts

```diff
@@ -0,0 +1,6 @@
+import { User } from './user.interface';
+
+export interface ChallengeResponse {
```

Comment:
Would it be clearer to have

```suggestion
export interface EmailVerificationChallengeResponse {
```

Reply:
I'm using the ChallengeResponse for the password challenge response too (I could separate those but they are identical for now?).

Reply:
Yeah maybe separate types to denote that the response data are only incidentally and not purposely the same.

Reply:
separated!